struct.go

```diff
@@ -105,6 +106,12 @@ func structNewIndex(L *lua.LState) int {
 			hint := field.Type().Elem()
 			goValue := lValueToReflect(L, value, hint, nil)
 
+			if !goValue.IsValid() {
+				// Occurs if the assigned value does not match the expected
+				// type for the field
+				L.RaiseError(fmt.Sprintf("could not set field %s: expected type %v", key, hint))
```

Comment:
Also rather than "expected type %v", it might be better to phrase it as "could not convert value to %v", since of course we are not expecting that exact type, only something that can be converted to it.